Xamarin.Forms.Platform.UAP/FormsButton.cs

```diff
@@ -73,11 +75,13 @@ void UpdateBackgroundColor()
 
 		void UpdateBorderRadius()
 		{
+			var radius = BorderRadius == -1 ? 0 : BorderRadius;
+
 			if (_contentPresenter != null)
-			{
-				var radius = BorderRadius == -1 ? 0 : BorderRadius;
 				_contentPresenter.CornerRadius = new Windows.UI.Xaml.CornerRadius(radius);
-			}
+
+			if (_rootGrid != null)
```

Comment:
This should have the added benefit of getting rid of the square highlight behind Buttons with rounded corners. Should look to see if there are any open issues about this on UWP.